holmes/plugins/toolsets/kubernetes.yaml

```diff
@@ -1,177 +1,180 @@
+              for (i=4; i<=NF; i++) {
+                requests = requests " " $i;
+              }
+              print namespace, name, sum_memory(requests) " Mi";
+            }' | sort -k3 -nr
+
+      - name: "kubectl_memory_requests_namespace"
+        description: "Fetch and display memory requests for all pods in a specified namespace in MiB, summing requests across multiple containers where applicable and handling binary, decimal, and millibyte units correctly."
+        command: |
+          kubectl get pods -n {{ namespace }} -o custom-columns="NAMESPACE:.metadata.namespace,NAME:.metadata.name,MEMORY_REQUEST:.spec.containers[*].resources.requests.memory" --no-headers | \
+          awk '
+            function convert_to_mib(value) {
+              if (value ~ /^[0-9]+e[0-9]+$/) return (value + 0) / (1024 * 1024); # Scientific notation
+              if (value ~ /m$/) return (value + 0) / (1024^2 * 1000);           # Millibytes (m)
+              if (value ~ /Ei$/) return (value + 0) * 1024^6 / (1024^2);        # Binary units
+              if (value ~ /Pi$/) return (value + 0) * 1024^5 / (1024^2);
+              if (value ~ /Ti$/) return (value + 0) * 1024^4 / (1024^2);
+              if (value ~ /Gi$/) return (value + 0) * 1024^3 / (1024^2);
+              if (value ~ /Mi$/) return (value + 0);
+              if (value ~ /Ki$/) return (value + 0) / 1024;
+              if (value ~ /E$/) return (value + 0) * 1000^6 / (1024^2);         # Decimal units
+              if (value ~ /P$/) return (value + 0) * 1000^5 / (1024^2);
+              if (value ~ /T$/) return (value + 0) * 1000^4 / (1024^2);
+              if (value ~ /G$/) return (value + 0) * 1000^3 / (1024^2);
+              if (value ~ /M$/) return (value + 0) * 1000^2 / (1024^2);
+              if (value ~ /k$/) return (value + 0) * 1000 / (1024^2);
+              return (value + 0) / (1024 * 1024);                               # Default: bytes
+            }
+            function sum_memory(requests) {
+              gsub(/^[ \t]+|[ \t]+$/, "", requests);
+              if (requests == "" || requests == "<none>") return 0;
+              split(requests, arr, ",");
+              total = 0;
+              for (i in arr) {
+                if (arr[i] != "<none>") total += convert_to_mib(arr[i]);
+              }
+              return total;
+            }
+            {
+              namespace = $1;
+              name = $2;
+              requests = $3;
+              for (i=4; i<=NF; i++) {
+                requests = requests " " $i;
+              }
+              print namespace, name, sum_memory(requests) " Mi";
+            }' | sort -k3 -nr
+
+      - name: "kubectl_events"
+        description: "Retrieve the events for a specific Kubernetes resource. `resource_type` can be any kubernetes resource type: 'pod', 'service', 'deployment, 'job'', 'node', etc."
+        command: "kubectl events --for {{resource_type}}/{{ pod_name }} -n {{ namespace }}"
```

Comment:
This above is the new event. Other changes are just indentation.

Reply:
Thanks.